posthog/api/paths.py

```diff
@@ -22,34 +24,40 @@ def _determine_path_type(self, request):
         
         # Default
         event: Optional[str] = "$pageview"
+        event_filter = {"event":event}
         path_type = "properties->> \'$current_url\'"
 
         # determine requested type
         if requested_type:
             if requested_type == "$screen":
                 event = "$screen"
+                event_filter = {"event":event}
                 path_type = "properties->> \'$screen_name\'"
             elif requested_type == "$autocapture":
                 event = "$autocapture"
+                event_filter = {"event":event}
                 path_type = "tag_name_source"
             elif requested_type == "custom_event":
                 event = None
+                event_filter = {'event__regex':'^[^\$].*'}
                 path_type = "event"
-        return event, path_type
+        return event, path_type, event_filter
 
     # FIXME: Timestamp is timezone aware timestamp, date range uses naive date.
     # To avoid unexpected results should convert date range to timestamps with timezone.
     def list(self, request):
         team = request.user.team_set.get()
         resp = []
         date_query = request_to_date_query(request.GET)
-        event, path_type = self._determine_path_type(request)
+        event, path_type, event_filter = self._determine_path_type(request)
+        properties = request.GET.get('properties')
 
-        sessions = Event.objects.filter(
+        sessions = Event.objects.add_person_id(team.pk).filter(
```

Comment:
was slightly concerned about this but it didn't have any meaningful effect on query time from eyeballing so i didn't condition it out

Reply:
Django ORM only actually adds the person ID when required (eg when filtering on person property)